### DISCUSSION TITLE
How to ignore folders?

### DISCUSSION THREAD

**Discussion by psociety** [webpack/webpack#16557] [Q&A] [answered]
Hi!
How can i force webpack/webpack-dev-middleware to ignore a folder?
I have a 700mb folder stored in the static files folder and for some reason webpack seems to try to bundle and watch all files inside it, which takes +4min when i do `npm start` as it's stuck on `wait until bundle finished /`.

The 700mb folder is `assets/resources`.
This is my `webpack.config.js`:
```js
const HtmlWebpackPlugin = require('html-webpack-plugin');
const path = require('path');

const CONFIG = {
  mode: 'development',
  entry: './src/index.ts',
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        include: path.resolve(__dirname, 'src'),
      },
    ],
  },
  watchOptions: {
    ignored: [path.resolve(__dirname, 'node_modules'), path.resolve(__dirname, 'assets/resources')],
  },
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
  },
  devServer: {
    port: 8080,
    static: {
      directory: path.join(__dirname, 'assets'),
      publicPath: '/assets',
    },
  },
  plugins: [
    new HtmlWebpackPlugin({
        template: './assets/index.html',
        filename: 'index.html',
        inject: false
    })
  ],
  output: {
    filename: 'main.js',
    path: path.resolve(__dirname, 'dist'),
  },
};

// This line enables bundling against src in this repo rather than installed module
//module.exports = env => (env ? require('../../../webpack.config.local')(CONFIG)(env) : CONFIG);
module.exports = env => { return CONFIG };
````

I also tried adding `watch: false`, but it makes no difference.

`packages.json`:
```json
"devDependencies": {
    "html-webpack-plugin": "^5.3.2",
    "speed-measure-webpack-plugin": "^1.5.0",
    "ts-loader": "^9.2.5",
    "typescript": "^4.4.2",
    "webpack": "^5.51.1",
    "webpack-cli": "^4.8.0",
    "webpack-dev-server": "^4.0.0"
  }
```

If i remove the folder, webpack takes less than a second to start. So that's clearly the problem, what i dont know i how to tell it to stop looking at that folder :(

**Comment by alexander-akait** [comment, marked as answer]
But you wan to serve `directory: path.join(__dirname, 'assets')`, so server takes a time to start

**Reply by psociety** [reply]
but i don't understand, why it "scans" the folder in order to serve it? can't it just act like any static server/nginx and just read the files when needed?

**Reply by alexander-akait** [reply]
I think middleware takes time due scan of list of files and other things, hard to say, try to set `static: false` to check this problem

**Reply by psociety** [reply]
it went from 4min to 21 secs. What does static exactly do that it requires scanning? : s

Edit: Ah my friend, it does also have "watching" -> https://webpack.js.org/configuration/dev-server/#watch
Thank you so much @alexander-akait ! You guided me to find the offending watcher!
